psi4/driver/p4util/fcidump.py

```diff
@@ -0,0 +1,209 @@
+        for n in range(frzcpi[h], frzcpi[h] + active_mopi[h]):
+            orbsym += '{:d},'.format(h + 1)
+            if not wfn.same_a_b_orbs():
+                orbsym += '{:d},'.format(h + 1)
+    header += 'ORBSYM={}\n'.format(orbsym)
+    header += '&END\n'
+    with open(fname, 'w') as intdump:
+        intdump.write(header)
+
+    # Get an IntegralTransform object
+    check_iwl_file_from_scf_type(core.get_option('SCF', 'SCF_TYPE'), wfn)
+    spaces = [core.MOSpace.all()]
+    transType = core.IntegralTransform.TransformationType.Restricted
+    if not wfn.same_a_b_orbs():
+        transType = core.IntegralTransform.TransformationType.Unrestricted
+    ints = core.IntegralTransform(wfn, spaces, transType)
+    ints.transform_tei(core.MOSpace.all(), core.MOSpace.all(), core.MOSpace.all(), core.MOSpace.all())
+    core.print_out('Integral transformation complete!\n')
+
+    DPD_info = {'instance_id': ints.get_dpd_id(), 'alpha_MO': ints.DPD_ID('[A>=A]+'), 'beta_MO': 0}
+    if not wfn.same_a_b_orbs():
+        DPD_info['beta_MO'] = ints.DPD_ID("[a>=a]+")
+    # Write TEI to fname in FCIDUMP format
+    core.fcidump_tei_helper(nirrep, wfn.same_a_b_orbs(), DPD_info, ints_tolerance, fname)
+
+    # Read-in OEI and write them to fname in FCIDUMP format
+    # Indexing functions to translate from zero-based (C and Python) to
+    # one-based (Fortran)
+    mo_idx = lambda x: x + 1
+    alpha_mo_idx = lambda x: 2 * x + 1
+    beta_mo_idx = lambda x: 2 * (x + 1)
+
+    with open(fname, 'a') as intdump:
+        core.print_out('Writing frozen core operator in FCIDUMP format to ' + fname + '\n')
+        if reference == 'RHF':
+            PSIF_MO_FZC = 'MO-basis Frozen-Core Operator'
+            moH = core.Matrix(PSIF_MO_FZC, wfn.nmopi(), wfn.nmopi())
+            moH.load(core.IO.shared_object(), constants.PSIF_OEI)
+            mo_slice = core.Slice(frzcpi, active_mopi)
+            MO_FZC = moH.get_block(mo_slice, mo_slice).to_array()
+            offset = 0
+            for h, block in enumerate(MO_FZC):
+                il = np.tril_indices(block.shape[0])
+                for index, x in np.ndenumerate(block[il]):
+                    row = mo_idx(il[0][index] + offset)
+                    col = mo_idx(il[1][index] + offset)
+                    intdump.write('{:29.20E} {:4d} {:4d} {:4d} {:4d}\n'.format(x, row, col, 0, 0))
+                offset += block.shape[0]
+            # Additional one-electron integrals as requested in oe_ints
+            # Orbital energies
+            core.print_out('Writing orbital energies in FCIDUMP format to ' + fname + '\n')
+            if 'EIGENVALUES' in oe_ints:
+                iorb = 0
+                eps_a = wfn.epsilon_a().get_block(mo_slice).to_array()
+                for h, block in enumerate(eps_a):
+                    for idx, x in np.ndenumerate(block):
+                        intdump.write('{: 29.20E} {:4d} {:4d} {:4d} {:4d}\n'.format(x, mo_idx(iorb), 0, 0, 0))
```

Comment:
It might be worth using the native NumPy writers for speed. Not sure you can make them jump through the right hoops, but perhaps worth considering: https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.savetxt.html